src/Actions/FindExtensionFromHeadersAction.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Strictphp\HttpClients\Actions;
+
+use Psr\Http\Message\ResponseInterface;
+use Strictphp\HttpClients\Contracts\FindExtensionFromHeadersActionContract;
+
+final class FindExtensionFromHeadersAction implements FindExtensionFromHeadersActionContract
+{
+    public function execute(ResponseInterface $response): string
+    {
+        $headers = $response->getHeader('Content-Type');
+
+        foreach ($headers as $value) {
+            $lower = strtolower($value);
+            $extension = match (true) {
+                str_contains($lower, 'json') => 'json',
+                str_contains($lower, 'html') => 'html',
+                str_contains($lower, 'pdf') => 'pdf',
+                default => static::more($lower),
+            };
+            if ($extension !== null) {
+                return $extension;
+            }
+        }
+
+        return 'txt';
+    }
+
+    protected static function more(string $extension): ?string
```

Comment:
k čemu je toto?

Reply:
možnost rozšířit, když by jsi chtěl doplnit převod nějakého content type na konkrétní koncovku, aby jsi to nemusel celé implementovat ale podědil a upravil metodu... já má třídu final?